src/models/Domain.ts

```diff
@@ -89,25 +86,29 @@ export default class Domain extends FastlyCachableModel {
   }
 
   protected async beforeValidate(): Promise<void> {
-    if (!this.parent) {
-      this.parent = await this.setParent();
+    if (!this.parentId) {
+      this.parentId = (await this.setParent())?.id || null;
     }
   }
 
   private async setParent(): Promise<Domain | null> {
-    const splittedName = this.getSplittedName();
-    let parentName = splittedName[splittedName.length - 1];
+    if (this.name.indexOf('.') > 0) {
```

Comment:
lets make the function more flat:

```
```suggestion
    if (this.name.indexOf('.') === 0) {
      return null;
    }
    ...
```

Reply:
worth to make this a util in utils/domain?
something like getParentDomain?

Reply:
Refactored and added some tests.
I don't think it's worth it to make this a util since in general you would just use the `parent` relation to find the parent. This function is only necessary specifically to fill in the parent id before saving the entity.